compiler-core/src/language_server/annotation_printer.rs

```diff
@@ -0,0 +1,218 @@
+#![allow(dead_code)]
+
+use ecow::EcoString;
+use im::HashMap;
+use itertools::Itertools;
+use std::sync::Arc;
+
+use crate::{
+    ast::TypeAst,
+    type_::{Type, TypeVar, PRELUDE_MODULE_NAME},
+};
+
+#[derive(Debug)]
+pub struct AnnotationPrinter<'a> {
+    names: HashMap<u64, EcoString>,
+    uid: u64,
+
+    // A mapping of printd type names to the module that they are defined in.
+    printed_types: HashMap<EcoString, EcoString>,
+
+    // Mapping of module.type to type alias -
+    // eg: import mod1.{type Cat as c} -> Key: (mod1, Cat), Value: c
+    // eg: type UserId = Int -> Key: (mod, UserId), Value: Int
```

Comment:
Oh the field `type_aliases` contains mapping of pairs of types and the module in which they are defined to their alias in the current module. (Sorry the second example is actually flipped since the value has to be UserId).  

So for example, lets say there is an import statement in a module like so `import mod1.{type Tiger as Tig}`. This would result in a map entry with the key being ("mod1", "Tiger") and the value being Tig. -> `type_aliases.insert(("mod1", "Tiger"), "Tig")`

Another example is that of an type alias statement like so in a module named foo: `type UserId = Int`. Since `Int` has been aliased by `UserId`,  an entry in the map is inserted, with they key being a pair of (foo, Int) and the value being UserId. (`type_aliases.insert(("foo", "Int"), "UserId"`). So any occurrences of the above types in a module which are not already annotated by the user, could be annotated with their aliased types.

Reply:
I think that type alias one is going to break for generics

Reply:
Hmm sorry, not quite sure on how this would break generics. Could you give me an example ?